napari/layers/utils/spatial_information.py

```diff
@@ -0,0 +1,331 @@
+    TYPE_CHECKING,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    overload,
+)
+
+import numpy as np
+import pint
+from psygnal import Signal, SignalGroup
+
+from napari.layers.utils.layer_utils import coerce_affine
+from napari.utils.transforms import Affine, CompositeAffine, TransformChain
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+
+    UnitsLike = Union[None, str, pint.Unit, dict[str, Union[str, pint.Unit]]]
+    UnitsInfo = Union[None, pint.Unit, dict[str, pint.Unit]]
+
+
+__all__ = ('SpatialInformation',)
+_OPTIONAL_PARAMETERS = {
+    'affine',
+    'axes_labels',
+    'rotate',
+    'scale',
+    'shear',
+    'translate',
+    'units',
+}
+
+
+class SpatialInformationEvents(SignalGroup):
+    affine = Signal(Affine)
+    axes_labels = Signal(tuple)
+    rotate = Signal(np.ndarray)
+    scale = Signal(tuple)
+    shear = Signal(np.ndarray)
+    translate = Signal(tuple)
+    units = Signal(dict)
+
+
+class SpatialInformation:
+    """Spatial information for layers.
+
+    Parameters
+    ----------
+    ndim: int, optional
+        Number of dimensions to be represented.
+    affine : array-like or napari.utils.transforms.Affine, optional
+        An existing affine transform object or an array-like that is its transform matrix.
+    axes_labels : Sequence[str], optional
+        Sequence of length `ndim` containing the labels for each axis.
+    rotate : float, 3-tuple of float, or n-D array, optional
+        If a float convert into a 2D rotation matrix using that value as an
+        angle. If 3-tuple convert into a 3D rotation matrix, using a yaw,
+        pitch, roll convention. Otherwise assume an nD rotation. Angles are
+        assumed to be in degrees. They can be converted from radians with
+        np.degrees if needed.
+    scale : tuple of float, optional
+        Scale factors for the layer. Length of scale should be equal to ndim.
+    shear : 1-D array or n-D array, optional
+        Either a vector of upper triangular values, or an nD shear matrix with
+        ones along the main diagonal.
+    translate : tuple of float, optional
+        Translation values for the layer. Length of translate should be equal to ndim.
+    units :
+
+    scale : tuple of float
+        Scale factors for the layer.
+
+
+    Attributes
+    ----------
+    affine : napari.utils.transforms.Affine
+        Affine transform object.
+
+
+    """
+
+    def __init__(
+        self,
+        *,
+        ndim: int,
+        affine: Union[npt.ArrayLike, Affine, None] = None,
+        axes_labels: Optional[Sequence[str]] = None,
+        rotate: Union[
+            float, Tuple[float, float, float], npt.ArrayLike, None
+        ] = None,
+        scale: Optional[Sequence[float]] = None,
+        shear: Optional[npt.ArrayLike] = None,
+        translate: Optional[Sequence[float]] = None,
+        units: UnitsLike = None,
+    ):
+        locals_dict = locals()
+        self._unset_parameters = {
```

Comment:
This only seems to be used in testing.

If there has another purpose in the future (and I think it should!) then you should add some comments to the code to explain that purpose.

Reply:
It is to allow inheriting spatial information from existing layers. To know if spacing `(1, 1, 1)` is the default value or value provided by the layer creator.